x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/BaseDateTimeProcessor.java

```diff
@@ -42,12 +44,19 @@ public Object process(Object input) {
             return null;
         }
 
-        if (!(input instanceof ZonedDateTime)) {
-            throw new SqlIllegalArgumentException("A date is required; received {}", input);
+        if (input instanceof ZonedDateTime) {
+            return doProcess(((ZonedDateTime) input).withZoneSameInstant(zoneId));
+        }
+        if (input instanceof OffsetTime) {
+            return doProcess(((OffsetTime) input).withOffsetSameInstant(ZoneOffset.UTC));
```

Comment:
Why should the time be converted to UTC? The zoneId of the function should be used instead (potentially by converting it to a ZoneOffset or going through an Instant before).

Reply:
That's wrong, fixed and added tests.

Reply:
yep, that's wrong